plugins/redis/user_credentials.go

```diff
@@ -0,0 +1,84 @@
+package redis
+
+import (
+	"fmt"
+
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func UserCredentials() schema.CredentialType {
+	return schema.CredentialType{
+		Name:    credname.UserCredentials,
+		DocsURL: sdk.URL("https://redis.io/docs/ui/cli/#host-port-password-and-database"),
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.Password,
+				MarkdownDescription: "Password used to authenticate to Redis server.",
+				Secret:              true,
+				Composition: &schema.ValueComposition{
+					Length: 32,
+					Charset: schema.Charset{
+						Uppercase: true,
+						Lowercase: true,
+						Digits:    true,
+					},
+				},
+			},
+			{
+				Name:                fieldname.Username,
+				MarkdownDescription: "Username used to authenticate to Redis server.",
+				Secret:              false,
+				Optional:            true,
+				Composition: &schema.ValueComposition{
+					Charset: schema.Charset{
+						Uppercase: true,
+						Lowercase: true,
+						Digits:    true,
+					},
+				},
+			},
+			{
+				Name:                fieldname.Host,
+				MarkdownDescription: "Host address for the Redis server.",
+				Secret:              false,
+				Optional:            true,
+				Composition: &schema.ValueComposition{
+					Charset: schema.Charset{
+						Lowercase: true,
+						Symbols:   true,
+						Digits:    true,
+					},
+				},
+			},
+			{
+				Name:                fieldname.Port,
+				MarkdownDescription: "Port for the Redis server.",
+				Secret:              false,
+				Optional:            true,
+				Composition: &schema.ValueComposition{
+					Charset: schema.Charset{
+						Digits: true,
+					},
+				},
+			},
+		},
+		DefaultProvisioner: provision.EnvVars(defaultEnvVarMapping),
+		Importer: importer.TryAll(
+			importer.TryEnvVarPair(defaultEnvVarMapping),
+		)}
+}
+
+var defaultEnvVarMapping = map[string]sdk.FieldName{
+	"REDISCLI_AUTH": fieldname.Password,
+}
+
+var argsToProvision = []string{
+	fmt.Sprintf("--user %s", fieldname.Username),
+	fmt.Sprintf("-h %s", fieldname.Host),
+	fmt.Sprintf("-p %s", fieldname.Port),
+}
```

Comment:
We're looking to provision the _value_ here, right? Not the name. 🤔

Reply:
Also, the flag name and the value should be separate args.

Reply:
Probably simpler to set the args directly in the provisioner, doesn't make much sense anyway to split it out until we merge your args provisioner PR.